src/becca/entities/abstract_becca_entity.ts

```diff
@@ -70,31 +84,33 @@ class AbstractBeccaEntity {
         return utils.hash(contentToHash).substr(0, 10);
     }
 
-    /** @protected */
-    getPojoToSave() {
+    protected getPojoToSave() {
         return this.getPojo();
     }
 
-    /**
-     * @protected
-     * @abstract
-     */
-    getPojo() {
-        throw new Error(`Unimplemented getPojo() for entity '${this.constructor.name}'`)
+    hasStringContent(): boolean {
+        // TODO: Not sure why some entities don't implement it.
+        return true;
+    }
+
+    abstract getPojo(): {};
+
+    get isDeleted(): boolean {
+        // TODO: Not sure why some entities don't implement it.
+        return false;
     }
 
     /**
      * Saves entity - executes SQL, but doesn't commit the transaction on its own
-     *
-     * @returns {this}
      */
-    save(opts = {}) {
-        const entityName = this.constructor.entityName;
-        const primaryKeyName = this.constructor.primaryKeyName;
+    save(): this {
+        const constructorData = (this.constructor as unknown as ConstructorData<T>);
```

Comment:
@ProfDoof , it feels weird because it is weird.

This is a pattern in the original code that I couldn't wrap my head around it initially. It seems that the original code uses `this.constructor.*` to store constants at class-level, from the base class (`AbstractBeccaEntity`) to the actual concrete class..

So all `BAttribute`s will share a single `ConstructorData`, `BBranch` another one and so on.

`ConstructorData` is an interface that contains all the data stored in `constructor`. I'll add a few comments but I won't refactor it here since it's outside the scope of the PR.